pkg/controllers/work/apply_controller.go

```diff
@@ -729,13 +655,18 @@ func computeManifestHash(obj *unstructured.Unstructured) (string, error) {
 
 // isManifestManagedByWork determines if an object is managed by the work controller.
 func isManifestManagedByWork(ownerRefs []metav1.OwnerReference) bool {
```

Comment:
@ryanzhang-oss made some changes to the existing behavior. Please let me know if it's good.